src/components/functional/PaymentRequestDetailsModal/PaymentRequestDetailsModal.tsx

```diff
@@ -0,0 +1,136 @@
+import { useEffect, useState } from 'react';
+import UIPaymentRequestDetailsModal from '../../ui/PaymentRequestDetailsModal/PaymentRequestDetailsModal';
+import { parseApiTagToLocalTag } from '../../../utils/parsers';
+import { LocalPaymentRequest, LocalTag } from '../../../types/LocalTypes';
+import { PaymentRequestClient } from '../../../api/clients/PaymentRequestClient';
+import { PaymentRequest, PaymentRequestUpdate } from '../../../api/types/ApiResponses';
+import { MerchantClient } from '../../../api/clients/MerchantClient';
+
+interface PaymentRequestDetailsModalProps {
+  token: string; // Example: "eyJhbGciOiJIUz..."
+  apiUrl: string; // Example: "https://api.nofrixion.com/api/v1"
+  merchantId: string;
+  selectedPaymentRequestID: string;
+  merchantTags: LocalTag[];
+  paymentRequests: LocalPaymentRequest[];
+  open: boolean;
+  onDismiss: () => void;
+  setMerchantTags: (merchantTags: LocalTag[]) => void;
+  setPaymentRequests: (paymentRequests: LocalPaymentRequest[]) => void;
+}
+const PaymentRequestDetailsModal = ({
+  token,
+  apiUrl,
+  merchantId,
+  selectedPaymentRequestID,
+  merchantTags,
+  paymentRequests,
+  open,
+  onDismiss,
+  setMerchantTags,
+  setPaymentRequests,
+}: PaymentRequestDetailsModalProps) => {
+  const paymentRequestClient = new PaymentRequestClient(apiUrl, token, merchantId);
+  const merchantClient = new MerchantClient(apiUrl, token, merchantId);
+
+  const [paymentRequest, setPaymentRequest] = useState<LocalPaymentRequest | undefined>(undefined);
+
+  useEffect(() => {
+    if (selectedPaymentRequestID) {
+      const paymentRequest = paymentRequests.find((paymentRequest) => paymentRequest.id === selectedPaymentRequestID);
+      if (paymentRequest) {
+        setPaymentRequest(paymentRequest);
+      }
+    }
+  }, [selectedPaymentRequestID, paymentRequests]);
+
+  const onRefundClick = async (paymentAttemptID: string) => {
+    //TODO: Will implement refund for atleast card payment attempts. For PISP, it will need to be worked on later.
+    console.log(paymentAttemptID);
+  };
+
+  const updatePaymentRequests = (updatedPaymentRequest: PaymentRequest) => {
+    for (let i = 0; i < paymentRequests.length; i++) {
```

Comment:
You can probably use paymentRequests.findIndex instead.

Reply:
Thanks! This is nice!